pandas/core/indexes/base.py

```diff
@@ -763,6 +763,17 @@ def get_values(self):
         """
         return self.values
 
+    @property
+    def _take_values(self):
+        """
+        Values to use in `take` operation, suitable to being passed back to
+        _shallow_copy/_simple_new.
+        """
+        if is_period_dtype(self):
+            # Avoid casting to object-dtype
+            return self._ndarray_values
+        return self.values
```

Comment:
@TomAugspurger @jorisvandenbossche is there a better name for this that might fit with EA naming conventions?

@jschendel should IntervalIndex be special-cased here?

Reply:
> @jschendel should IntervalIndex be special-cased here?

The implementation should be good as-is for `IntervalIndex`; `IntervalIndex.values` is an `IntervalArray`, which has a special-cased `take` method.  `IntervalIndex._shallow_copy/_simple_new` should accept an `IntervalArray` (which we'd get from `take`), so should be good on that front.

Reply:
this is basically .asi8 on Indexes now, yes? Is there a reason we are not aligning on that? or changing .asi8 to this?